Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1062 @@
+            $NonCompliantFrontDoorEndpoints | Export-CSV -Path $BackupFile -NoTypeInformation
+            Write-Host "Front Doors details have been backed up to [$($BackupFile)]." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        Write-Host "Minimum required TLS Version [" $($requiredMinTLSVersion) "] will be set on all custom domains of front doors." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+
+        # Here AutoRemediation switch is used as there is no need to take user input at BRS level if user has given consent to proceed with the remediation in AutoRemediation Script.
+        if(-not $AutoRemediation)
+        {
+            if (-not $Force)
+            {
+                Write-Host "Do you want to set minimum required TLS Version for all Custom Domains? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+                
+                $userInput = Read-Host -Prompt "(Y|N)"
+                Write-Host $([Constants]::SingleDashLine)
+                if($userInput -ne "Y")
+                {
+                    Write-Host "Minimum required TLS Version will not be set for any Custom Domain. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+                    Write-Host $([Constants]::SingleDashLine)
+                    return
+                }
+                Write-Host "User has provided consent to set minimum required TLS Version on Custom Domains for all Front Doors." -ForegroundColor $([Constants]::MessageType.Update)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+            else
+            {
+                Write-Host "'Force' flag is provided. Minimum required TLS Version will be set on the for all Front Doors without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "`[Step 4 of 4] Set minimum required TLS Version for FrontDoors"
+        Write-Host $([Constants]::SingleDashLine)
+        # To hold results from the remediation.
+        $customDomainsRemediated = @()
+        $frontDoorsEPSkipped = @()
+        
+        $NonCompliantFrontDoorEndpoints | ForEach-Object {
+            $frontdoorEndpoint = $_
+            $resourceGroupName = $_.ResourceGroupName
+            $resourceName = $_.ResourceName
+            $DomainName = $_.DomainName
+            $CertificateType =$_.CertificateType
+            $PrevMinimumTlsVersion = $_.MinimumTlsVersion
+            $MinimumTlsVersion = $_.MinimumTlsVersion
+            $isMinTlsVersionSetOnCustomDomain = $_.isMinTLSVersionSetOnCustomDomain
+            $isMinTLSVersionSetOnCustomDomainPostRemediation = $_.MinimumTlsVersion
```

Comment:
From naming convention it looks like a boolean, but we are assigning minTLS version to it, please check if it is expected or not?

Reply:
Not expected I have assigned it a bool value